src/chainstate/burn/operations/leader_key_register.rs

```diff
@@ -174,7 +174,14 @@ impl LeaderKeyRegisterOp {
             }
         };
 
-        let address = outputs[0].address.clone();
+        let address = outputs[0]
+            .address
+            .clone()
+            .try_into_stacks_address()
+            .ok_or_else(|| {
+                test_debug!("Invalid address: not representable as a StacksAddress");
+                op_error::InvalidInput
+            })?;
```

Comment:
Is this a consensus change?

Reply:
Nope. First, `.address` isn't even used in `LeaderKeyRegister`. I'm even going so far as to remove it in my segwit PR.  Second, `.try_into_stacks_address()` is infallible right now, since the only `PoxAddress` variant (`Standard(addr)`) contains a `StacksAddress`.

Reply:
In that case, this should panic -- as it is currently written, it suggests that an error would invalidate the `LeaderKeyRegister` (similarly for all the operations that this `try_into_stacks_address()` method is invoked on).

Reply:
The reason it's written this way is because `.try_into_stacks_address()` will soon be fallible, i.e. on the arrival of the native segwit support.  The reason it will be fallible is because there will not be a way to convert a segwit address into a `principal` type, so parsing `StackStxOp` and `TransferStxOp` must fail here if the intended recipient can't be represented as a `principal`.